### PR TITLE
First draft of Synthetic PoRep (FIP-0059)

### PR DESCRIPTION
**Pull request by irenegia**
FIP draft for the synthetic porep protocol
cc @lucaniz @nicola @Kubuxu

### CODE REVIEW

File: fip-00xx.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    1. We assume there is a sector S for which the Storage Provider completed PreCommit1 and PreCommit2 computations.
+    2. This sector is not listed on-chain.
+    3. The Storage Provider possesses knowledge or CommR and CommD of that sector (acquired in PreCommit2 step) and the layers needed to generate CommR
+2. **Storage Provider generates “Synthetic” challenges from CommR**
+    1. Based on the CommR, the Storage Provider generates a list of `SynthChallengesNumber` challenges
+    2. The Storage Provider computes responses for all the `SynthChallengesNumber` challenges, which take the form of `SynthChallengesNumber` vanilla proofs and saves them for future use.
+3. **Storage Provider can remove layers data**
+    1. As the Storage Provider knows responses to all possible challenges that will be asked in the interactive step, he can remove the layers data which is needed to respond to challenges.
+4. **Storage Provider publishes “PreCommitsSector”**
+    1. Using the same flow as today Storage Provider submits the sector for PreCommit.
```
Comment (anorth) [inline fip-00xx.md:36]: Is this exactly the same PreCommit method/s used today, or do we need a new parameter to indicate that synthetic PoRep is to be used at proof time?

Comment (Kubuxu) [reply fip-00xx.md:36]: We would use a new PoRep version/type. The handling on the side of actors is equivalent between interactive and synthetic-interactive PoReps.

**Review by kaitlin-beegle — APPROVED** [review]

**Comment by anorth** [comment]
Just noting that I don't think this FIP is ready for final consideration yet. The specification is not yet detailed enough to implement. E.g., at a minimum it must specify a new PoRep proof type, and then how that new type is to be used in code. The current specification is more of a workflow description, but we need a spec of the functional changes to built-in actor code (and any other consensus-relevant pieces). 

This can still merge, but it's not ready to go yet.

**Comment by irenegia** [comment]
@anorth details added in the spec section! 
Can we get your approval now? Thanks! :)

**Review by anorth — APPROVED** [review]

**Pull request merged**